src/main/java/net/glowstone/block/blocktype/BlockRedstone.java

```diff
@@ -0,0 +1,121 @@
+                    if (target.getType().isSolid() && target.getRelative(BlockFace.DOWN).getType() == Material.REDSTONE_TORCH_ON) {
+                        if (me.getData() != 15) {
+                            me.setData((byte) 15);
+                        }
+                        return;
+                    }
+            }
+        }
+
+        byte power = 0;
+
+        for (BlockFace face : calculateConnections(me)) {
+            GlowBlock target = me.getRelative(face);
+            if (target.getType() != Material.REDSTONE_WIRE) {
+                if (!target.getType().isSolid()) {
+                    target = target.getRelative(BlockFace.DOWN);
+                } else if (!me.getRelative(BlockFace.UP).getType().isSolid()) {
+                    target = target.getRelative(BlockFace.UP);
+                }
+
+                if (target.getType() != Material.REDSTONE_WIRE) {
+                    // There is no redstone wire here..
+                    continue;
+                }
+            }
+
+            if (target.getData() > power) {
+                power = (byte) (target.getData() - 1);
+            }
+        }
+
+        if (power != me.getData()) {
+            me.setData(power);
+        }
+    }
+
+    private static final BlockFace[] ADJACENT = new BlockFace[]{BlockFace.NORTH, BlockFace.EAST, BlockFace.SOUTH, BlockFace.WEST, BlockFace.UP, BlockFace.DOWN};
+    private static final BlockFace[] SIDES = new BlockFace[]{BlockFace.NORTH, BlockFace.EAST, BlockFace.SOUTH, BlockFace.WEST};
+
+    public static List<BlockFace> calculateConnections(GlowBlock block) {
+        List<BlockFace> value = new ArrayList<>();
+        List<BlockFace> connections = new ArrayList<>();
+        value.add(BlockFace.DOWN);
+        for (BlockFace face : SIDES) {
+            GlowBlock target = block.getRelative(face);
+            switch (target.getType()) {
+                case REDSTONE_BLOCK:
+                case REDSTONE_TORCH_ON:
+                case REDSTONE_TORCH_OFF:
+                case REDSTONE_WIRE:
+                    connections.add(face);
+                    break;
+                default:
+                    if (target.getType().isSolid() && !block.getRelative(BlockFace.UP).getType().isSolid()
+                            && target.getRelative(BlockFace.UP).getType() == Material.REDSTONE_WIRE) {
+                        connections.add(face);
+                    } else if (!target.getType().isSolid()
+                            && target.getRelative(BlockFace.DOWN).getType() == Material.REDSTONE_WIRE) {
+                        connections.add(face);
+                    }
+                    break;
+            }
+        }
+
+        if (connections.isEmpty()) {
+            value.addAll(Arrays.asList(SIDES));
+        } else {
+            value.addAll(connections);
+            if (connections.size() == 1) {
+                value.add(connections.get(0).getOppositeFace());
+            }
+        }
+
```

Comment:
Connections is just for calculating, its not returned at all afterwards. Redstone dust visually doesn't turn off, for the player, this is a bug that I have fixed on my test server and I have not committed to this fork yet..